components/feature/session/src/main/java/mozilla/components/feature/session/ThumbnailsFeature.kt

```diff
@@ -0,0 +1,69 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.session
+
+import android.content.Context
+import android.support.annotation.VisibleForTesting
+import mozilla.components.browser.session.SelectionAwareSessionObserver
+import mozilla.components.browser.session.Session
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.concept.engine.EngineView
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import mozilla.components.support.ktx.android.content.isOSOnLowMemory
+
+/**
+ * Feature implementation for automatically taking thumbnails of sites.
+ */
```

Comment:
nit: Could be a bit more descriptive: When does it take thumbnails? What does it do with them?

Reply:
👍

Reply:
I didn't want to add when it takes the screenshot because I wasn't sure if it is an implementation detail and we may change it.
 
Does this sound better ? :)

```
/**
 * Feature implementation for automatically taking thumbnails of sites.
 * The feature will take a screenshot when the page finishes loading,
 * and will add it to the [Session.thumbnail] property.
 */
```

Reply:
Sounds great :)